src/core/ext/transport/chttp2/transport/writing.cc

```diff
@@ -609,15 +609,11 @@ grpc_chttp2_begin_write_result grpc_chttp2_begin_write(
     stream_ctx.FlushData();
     stream_ctx.FlushTrailingMetadata();
 
-    if (stream_ctx.stream_became_writable()) {
```

Comment:
we don't need to check for stream_became_writable anymore?

Reply:
```stream_became_writable``` is somewhat of a misnomer, it seems to mean a stream was unblocked by flow control for a message.

Reply:
I guess my question is - was it a mistake to be there earlier?

Reply:
Before ``` grpc_chttp2_end_write``` would only have an effect for streams that were unblocked by flow control, so it wasn't an issue, just kind of a misnomer of ```writable```.

Now we need to execute the new callback list, so we need to make sure it happens every time.

Reply:
hmm? but wait. this should again have the same problem. If we run the closures only on grpc_chttp2_begin_write, then it would only be called when there is another write or the previous write completed. Isn't it the same issue all over again?

Reply:
We flush the closures on both ```remove_stream``` and ```grpc_chttp2_end_write```.  In the cancellation case (the case that was broken before), remove_stream will be called, and that will take care of flushing the pending write ops.

Reply:
Added back this check.  PTAL